### PR TITLE
Added remote indexing for S3, Box and Http based WebDav remote URLs.

### PR DESCRIPTION
**Pull request by avasquez614**
Ticket craftercms/craftercms#2578

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -36,6 +36,7 @@
 		<!-- Override property setup by Spring Boot Parent -->
 		<jackson.version>2.9.7</jackson.version>
 		<aws.version>1.11.435</aws.version>
+		<box.sdk.version>2.23.0</box.sdk.version>
```
Comment (sumerjabri) [inline pom.xml:39]: Why not 2.24.0?

File: src/main/resources/base-target.yaml [hunk 2]
```diff
@@ -45,6 +45,20 @@ target:
           - application/vnd.ms-excel
           - application/vnd.ms-powerpoint
           - application/vnd.openxmlformats-officedocument.presentationml.presentation
+        # The regex path patterns for the metadata ("jacket") files of binary/document files
+        metadataPathPatterns:
+          - ^/?site/documents/.+\.xml$
+        # The regex path patterns for binary/document files that are store remotely
```
Comment (sumerjabri) [inline src/main/resources/base-target.yaml:51]: stored not store

File: src/main/resources/base-target.yaml [hunk 3]
```diff
@@ -45,6 +45,20 @@ target:
...
+        remoteBinaryPathPatterns: &remoteBinaryPathPatterns
```
Comment (sumerjabri) [inline src/main/resources/base-target.yaml:52]: Not all paths start with HTTP/HTTPS. Consider images/CSS/JS includes that simply start with `//` in order for the browser to use the current protocol in retrieving.

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/resources/base-target.yaml (hunk 3 above)
Comment (avasquez614) [reply src/main/resources/base-target.yaml:52]: Images, css and js are not indexed. The URLs in here are only for structured documents that are indexed. Binaries might not have been the best word tbh, but that's how structured documents are currently named in search classes. Should we change it?

Comment (sumerjabri) [reply src/main/resources/base-target.yaml:52]: Wouldn't that apply to "binaries" just as well? Let's discuss tomorrow.

**Pull request closed without merging**